source/blender/makesdna/DNA_sensor_types.h

```diff
@@ -189,8 +189,8 @@ typedef struct bJoystickSensor {
 	short axis_single;
 	int axisf;
 	int button;
-	int hat;
-	int hatf;
+	int hat; //unused
```

Comment:
Are they still used for compatibility conversion ? Else you can simply remove it.